week10_planning/seminar_MCTS.ipynb

```diff
@@ -6,46 +6,59 @@
    "source": [
     "## Seminar: Monte-carlo tree search (5 pts)\n",
     "\n",
-    "In this seminar, we'll implement a vanilla MCTS planning and use it to solve some Gym envs.\n",
+    "Monte Carlo tree search (MCTS) is a heuristic search algorithm, which shows cool results in challenging domains such as Go and chess. The algorithm builds a search tree, iteratively traverses it, and evaluates its nodes using a Monte-Carlo simulation.\n",
     "\n",
-    "![image.png](https://i.postimg.cc/6QmwnjPS/image.png)\n"
+    "In this seminar, we'll implement a MCTS([[1]](#1), [[2]](#2)) planning and use it to solve some Gym envs.\n",
+    "\n",
+    "![image.png](https://i.postimg.cc/6QmwnjPS/image.png)"
    ]
   },
   {
    "cell_type": "markdown",
    "metadata": {},
    "source": [
     "__How it works?__\n",
-    "\n",
-    "We just start with an empty tree and from there we iterate and expand it. There are several common procedures.\n",
+    "We just start with an empty tree and expand it. There are several common procedures.\n",
     "\n",
     "__1) Selection__\n",
+    "Starting from the root, recursively select the node that corresponds to the tree policy.  \n",
     "\n",
-    "Starting from the root, recursively select node with highest UCB-1 score - on the one hand, the most beneficial action, on the other hand, the least explored.\n",
+    "There are several options for tree policies, which we saw earlier as exploration strategies: epsilon-greedy, Thomson sampling, UCB-1. It was shown that in MCTS, UCB-1 achieves a good result. Further, we will consider the one, but you can try to use others.\n",
+    "What actions are chosen with UCB-1?. On one hand, this is the most beneficial action; on the other hand, it's the least explored one.\n",
```

Comment:
One last touch: you have a mismatch between "actions" (plural) and "this is the most beneficial..." (singular). How about this?

> Following the UCB-1 tree policy, we will choose an action that, on one hand, we expect to have the highest return, and on the other hand, we haven't explored much.

Reply:
Thanks again. Your option is cooler. =)